### ISSUE TITLE
App is crashing(not running on ios 18.0 and xcode 16)

### ISSUE THREAD

**Issue by Lazizbek97** [flutter/flutter#155403]
### Steps to reproduce

I have lately upgraded flutter version.(and ios 18, xcode 16)
Then the application which is working so far, started to crashing. 

Here i will put Podfile if helps:

```
platform :ios, '13.0'
install! 'cocoapods', :deterministic_uuids => false
# CocoaPods analytics sends network stats synchronously affecting flutter build latency.

project 'Runner', {
  'Debug' => :debug,
  'Profile' => :release,
  'Release' => :release,
}

def flutter_root
  generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
  unless File.exist?(generated_xcode_build_settings_path)
    raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
  end

  File.foreach(generated_xcode_build_settings_path) do |line|
    matches = line.match(/FLUTTER_ROOT\=(.*)/)
    return matches[1].strip if matches
  end
  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
end

require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)

flutter_ios_podfile_setup

target 'Runner' do
  use_frameworks!
  use_modular_headers!

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
  target 'RunnerTests' do
      inherit! :search_paths
    end
end

post_install do |installer|
  installer.aggregate_targets.each do |target|
    target.xcconfigs.each do |variant, xcconfig|
      xcconfig_path = target.client_root + target.xcconfig_relative_path(variant)
      IO.write(xcconfig_path, IO.read(xcconfig_path).gsub("DT_TOOLCHAIN_DIR", "TOOLCHAIN_DIR"))
    end
  end
  installer.pods_project.targets.each do |target|
  flutter_additional_ios_build_settings(target)
    target.build_configurations.each do |config|
      if config.base_configuration_reference.is_a? Xcodeproj::Project::Object::PBXFileReference
        xcconfig_path = config.base_configuration_reference.real_path
        IO.write(xcconfig_path, IO.read(xcconfig_path).gsub("DT_TOOLCHAIN_DIR", "TOOLCHAIN_DIR"))
      end
    end
  end
end
```

### Expected results

It should have worked as before.

### Actual results

Here is error log:

```
[!] CocoaPods could not find compatible versions for pod "GoogleUtilities/Logger":
  In snapshot (Podfile.lock):
    GoogleUtilities/Logger (= 7.13.3, ~> 7.12)

  In Podfile:
    firebase_core (from `.symlinks/plugins/firebase_core/ios`) was resolved to 3.5.0, which depends on
      Firebase/CoreOnly (= 11.0.0) was resolved to 11.0.0, which depends on
        FirebaseCore (= 11.0.0) was resolved to 11.0.0, which depends on
          GoogleUtilities/Logger (~> 8.0)


You have either:
 * changed the constraints of dependency `GoogleUtilities/Logger` inside your development pod `firebase_core`.
   You should run `pod update GoogleUtilities/Logger` to apply changes you've made.`
```

### Code sample

<details open><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[!] CocoaPods could not find compatible versions for pod "GoogleUtilities/Logger":
  In snapshot (Podfile.lock):
    GoogleUtilities/Logger (= 7.13.3, ~> 7.12)

  In Podfile:
    firebase_core (from `.symlinks/plugins/firebase_core/ios`) was resolved to 3.5.0, which depends on
      Firebase/CoreOnly (= 11.0.0) was resolved to 11.0.0, which depends on
        FirebaseCore (= 11.0.0) was resolved to 11.0.0, which depends on
          GoogleUtilities/Logger (~> 8.0)


You have either:
 * changed the constraints of dependency `GoogleUtilities/Logger` inside your development pod `firebase_core`.
   You should run `pod update GoogleUtilities/Logger` to apply changes you've made.
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.24.3, on macOS 15.0 24A335 darwin-arm64, locale en-UZ)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 16.0)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2023.1)
[✓] IntelliJ IDEA Ultimate Edition (version 2024.1)
[✓] VS Code (version 1.93.1)
[✓] Connected device (6 available)
[✓] Network resources

• No issues found!
```

</details>

**Comment by Lazizbek97** [comment]
i fixed it by putting: `$FirebaseSDKVersion = '10.29.0'`
on top of Podfile. But i don't think it's correct way

**Comment by danagbemava-nc** [comment]
Hi @Lazizbek97, this is caused by the firebase plugins as such should be reported in the flutterfire repo. A similar issue is in https://github.com/firebase/flutterfire/issues/13228

You may want to try resetting your local pods. See https://github.com/firebase/flutterfire/issues/13228#issuecomment-2329014686

Closing from here. 

Thank you
